sdks/cpp/src/agones/sdk.cc

```diff
@@ -20,15 +20,18 @@
 namespace agones {
 
 struct SDK::SDKImpl {
+  std::string host_;
   std::shared_ptr<grpc::Channel> channel_;
   std::unique_ptr<agones::dev::sdk::SDK::Stub> stub_;
   std::unique_ptr<grpc::ClientWriter<agones::dev::sdk::Empty>> health_;
   std::unique_ptr<grpc::ClientContext> health_context_;
 };
 
 SDK::SDK() : pimpl_{std::make_unique<SDKImpl>()} {
-  pimpl_->channel_ = grpc::CreateChannel("localhost:59357",
-                                         grpc::InsecureChannelCredentials());
+  const char* port = std::getenv("AGONES_SDK_GRPC_PORT");
+  pimpl_->host_ = "localhost:" + (port ? std::string(port) : "5937");
```

Comment:
Nit: It might be nice to pull the `std::string` construction outside the ternary expression. Consider this 

```cc
piple_->host_ = std::string("localhost:") + (port ? port : "59357");
```

Reply:
done.